lib/src/main/java/com/cloudinary/android/uploadwidget/ui/imagepreview/UploadWidgetImageView.java

```diff
@@ -0,0 +1,315 @@
+     *
+     * @param imageUri Uri of the image to be displayed.
+     */
+    public void setImageUri(Uri imageUri) {
+        this.imageUri = imageUri;
+    }
+
+    /**
+     * Start the cropping, showing the crop overlay fitted to the image bounds.
+     */
+    public void startCropping() {
+        cropOverlayView.setVisibility(VISIBLE);
+        isCropStarted = true;
+    }
+
+    /**
+     * Stop the cropping, hiding the crop overlay and setting the original bitmap.
+     */
+    public void stopCropping() {
+        isCropStarted = false;
+        setAspectRatioLocked(false);
+        cropOverlayView.setVisibility(INVISIBLE);
+
+        if (isCroppedBitmapDisplayed) {
+            setBitmap(getWidth(), getHeight());
+            isCroppedBitmapDisplayed = false;
+        }
+        if (rotationAngle != 0) {
+            rotateBitmapBy(-rotationAngle);
+        }
+        rotationAngle = 0;
+        imageView.setScaleType(ImageView.ScaleType.CENTER);
+        updateImageViewBitmap();
+    }
+
+    /**
+     * Whether it is during cropping or not.
+     *
+     * @return true if {@link UploadWidgetImageView#startCropping()} was called, and false if cropping was never started or {@link UploadWidgetImageView#stopCropping()} was called.
+     */
+    public boolean isCropStarted() {
+        return isCropStarted;
+    }
+
+    /**
+     * Get the current crop overlay's cropping points.
+     *
+     * @return Crop points that make the crop overlay diagonal.
+     */
+    public CropPoints getCropPoints() {
+        float ratio = (float) originalWidth / bitmap.getWidth();
+        if (rotationAngle % 180 != 0) {
+            ratio = (float) originalWidth / bitmap.getHeight();
+        }
+
+        CropPoints cropPoints = cropOverlayView.getCropPoints();
+        Point p1 = cropPoints.getPoint1();
+        Point p2 = cropPoints.getPoint2();
+        p1.x = (int) ((p1.x - bitmapBounds.left) * ratio);
+        p1.y = (int) ((p1.y - bitmapBounds.top) * ratio);
+        p2.x = (int) ((p2.x - bitmapBounds.left) * ratio);
+        p2.y = (int) ((p2.y - bitmapBounds.top) * ratio);
+
+        return cropPoints;
+    }
+
+    /**
+     * Crop the image, hiding the crop overlay.
+     */
+    public void cropImage() {
+        CropPoints cropPoints = cropOverlayView.getCropPoints();
+        Bitmap croppedBitmap = Bitmap.createBitmap(bitmap,
+                cropPoints.getPoint1().x - bitmapBounds.left,
+                cropPoints.getPoint1().y - bitmapBounds.top,
+                cropPoints.getPoint2().x - cropPoints.getPoint1().x,
+                cropPoints.getPoint2().y - cropPoints.getPoint1().y);
+
+        imageView.setImageBitmap(croppedBitmap);
+        imageView.setScaleType(ImageView.ScaleType.FIT_CENTER);
+        cropOverlayView.setVisibility(INVISIBLE);
+        isCroppedBitmapDisplayed = true;
+    }
+
+    /**
+     * Rotate the image by 90 degrees.
+     */
+    public void rotateImage() {
+        rotationAngle += 90 % 360;
```

Comment:
That should be `(rotationAngle += 90) % 360`.
`%` has precedence over `+=`, the right hand side will always be 90 (`90 % 360`) so`rotationAngle` will be cumulative and won't reset at 360.